test/fw/ptl/utils/pbs_logutils.py

```diff
@@ -562,6 +579,153 @@ def get_log_files(self, hostname, path, start, end, sudo=False):
+
+        for x in list_of_syslog_files:
+            run_cmd = [self.du.which(hostname, 'cat',
+                       level=logging.INFOCLI2), x, " | tail -", str(n)]
+            l_sys = self.du.run_cmd(hosts=hostname, cmd=run_cmd, sudo=True,
+                                    level=logging.DEBUG2)['out']
+            for l in l_sys:
+                if daemon_str in l:
+                    lines.append(l)
+
+        lines.sort(key=lambda x: datetime.strptime(
+            x[:15], "%b %d %H:%M:%S"))
+
+        return lines
+
+    def _get_rsyslog_files(self, hostname=None, severity=None, facility=None):
+        """
+        :param severity: syslog serverity in pbs.conf
+        :type severity: int
+        :param facility: syslog facilty in pbs.conf
+        :type facility: int
+        :return: list of priorites list[]
+        """
+        facility = int(facility)
+        severity = int(severity)
+        # PBS supported severity
+        p_severity = {
+            0: "emerg",
+            1: "alert",
+            2: "crit",
+            3: "err",
+            4: "warning",
+            5: "notice",
+            6: "info",
+            7: "debug",
+            -1: "*"
+        }
+        p_facility = {
+            1: "daemon",
+            2: "local0",
+            3: "local1",
+            4: "local2",
+            5: "local3",
+            6: "local4",
+            7: "local5",
+            8: "local6",
+            9: "local7",
+            10: "*"
+        }
+
+        list_of_syslog_files = set()
+
+        list_of_priority = {"*." + p_severity[severity],
+                            p_facility[facility] + ".*", "*.*"}
+
+        if severity is None and facility is None:
+            for x in p_severity:
+                for y in p_facility:
+                    c = p_facility[y] + "." + p_severity[x]
+                    list_of_priority.add(c)
+        elif severity is None and facility is not None:
+            for x in p_severity:
+                c = facility + "." + p_severity[x]
+                list_of_priority.add(c)
+        elif severity is not None and facility is None:
+            for x in p_severity:
+                if int(x) <= severity:
+                    for y in p_facility:
+                        c = p_facility[y] + "." + p_severity[x]
+                        list_of_priority.add(c)
+        else:
+            for x in p_severity:
+                if int(x) <= severity:
+                    c = p_facility[facility] + "." + p_severity[x]
+                    list_of_priority.add(c)
+                    c = "*." + p_severity[x]
+                    list_of_priority.add(c)
+
+        if hostname is None:
+            hostname = socket.gethostname()
+        f = self.du.cat(hostname, filename='/etc/rsyslog.conf',
+                        logerr=False, sudo=True)['out']
+
+        for line in f:
+            for p in list_of_priority:
+                if p in line and line[0] != '#':
```

Comment:
Can one priority have many message files? If not, then you can just break out of the inner loop when this if condition is true.